### PR TITLE
Persist state to printer_ui

### PR DESCRIPTION
**Pull request by weedgrease**
Sends `self.running` and current step title in GET requests. Also, cleaned up with pep8 guidelines.

### CODE REVIEW

File: octoprint_automation_scripts/__init__.py [hunk 1]
```diff
@@ -127,10 +134,11 @@ def mecode_entrypoint(self, script_id, extra_args):
             self.running = False
             self.active_script_id = None
             self._old_script_status = None
+            self.g = None
```
Comment (jminardi) [inline octoprint_automation_scripts/__init__.py:137]: Why did you have to add this?

File: octoprint_automation_scripts/__init__.py [hunk 2]
```diff
@@ -291,9 +303,11 @@ def serial_factory(self, comm_instance, port, baudrate, connection_timeout):
             comm_instance._changeState(comm_instance.STATE_DETECT_SERIAL)
             serial_obj = comm_instance._detectPort(True)
             if serial_obj is None:
-                comm_instance._errorValue = 'Failed to autodetect serial port, please set it manually.'
+                comm_instance._errorValue = "Failed to autodetect serial port,"
```
Comment (jminardi) [inline octoprint_automation_scripts/__init__.py]: I think you have to wrap this in parens to make it continue to the next line.

File: octoprint_automation_scripts/__init__.py [hunk 3]
```diff
@@ -322,7 +342,8 @@ def get_update_information(self, *args, **kwargs):
                 user="Voxel8",
                 repo="Octoprint-Automation-Scripts",
                 branch="master",
-                pip="https://github.com/Voxel8/OctoPrint-Automation-Scripts/archive/{target_version}.zip",
+                pip="https://github.com/Voxel8/OctoPrint-Automation-Scripts/"
```
Comment (jminardi) [inline octoprint_automation_scripts/__init__.py]: Same comment here about wrapping in parens

File: octoprint_automation_scripts/__init__.py [hunk 4]
```diff
@@ -413,7 +436,13 @@ def on_api_command(self, command, data):
             self.start(command, data)
 
     def on_api_get(self, request):
-        return flask.jsonify(script_titles=self.script_titles)
+        if self.active_script_id is not None:
```
Comment (jminardi) [inline octoprint_automation_scripts/__init__.py:439]: Is this block and the self.g addition the only real change in this PR?

**Review by jminardi — APPROVED** [review]
If self.g and on_api_get are the only real changes, then this PR is good to merge after you fix the string line continuation issues.

File: octoprint_automation_scripts/__init__.py (hunk 4 above)
Comment (weedgrease) [reply octoprint_automation_scripts/__init__.py:439]: Correct, yep!

File: octoprint_automation_scripts/__init__.py (hunk 3 above)
Comment (weedgrease) [reply octoprint_automation_scripts/__init__.py]: Fixed

File: octoprint_automation_scripts/__init__.py (hunk 2 above)
Comment (weedgrease) [reply octoprint_automation_scripts/__init__.py]: Fixed

**Pull request merged**